corehq/apps/users/management/commands/grant_web_apps_to_default_mobile_worker_role.py

```diff
@@ -0,0 +1,22 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.users.analytics import get_role_user_count
+from corehq.apps.users.models_role import Permission, UserRole
+
+
+class Command(BaseCommand):
+    help = "Add access_web_apps permission to default mobile worker roles"
+
+    def handle(self, **options):
+        permission, created = Permission.objects.get_or_create(value='access_web_apps')
+        roles_to_update = UserRole.objects.filter(is_commcare_user_default=True)
+        for role in roles_to_update:
+            permissions = role.permissions
+            if not permissions.access_web_apps:
```

Comment:
It would be cool to do this as a filter/exclude directly in the queryset, but I don't know how. I got to `roles_to_update.exclude(rolepermission__permission_fk_id=permission.id)` but that only checks if the foreign key exists, not the value of the permission.

Reply:
Would this work?
```py
roles = UserRole.objects.filter(
    is_commcare_user_default=True,
    rolepermission__permission_fk__value='access_web_apps',
).distinct()
```
It creates SQL like:
```sql
SELECT DISTINCT
  "users_userrole"."id",
  "users_userrole"."domain",
  "users_userrole"."name",
  "users_userrole"...
FROM "users_userrole"
INNER JOIN "users_rolepermission" ON ("users_userrole"."id" = "users_rolepermission"."role_id")
INNER JOIN "users_permission" ON ("users_rolepermission"."permission_fk_id" = "users_permission"."id")
WHERE ("users_userrole"."is_commcare_user_default" AND "users_permission"."value" = 'access_web_apps')
```
Which gets all `UserRole` rows with at least one related 'access_web_apps' permission.

Reply:
Oh, right, it probably is the same. I'm pretty sure we can do what you want, possibly with a subquery, and probably an even more efficient than what I suggested above. I'm happy to help you work on it if you think it's worth it.

Reply:
Let's revisit it if I run this on prod and it seems like it's going to take forever.